AzureFirewall/AzureFirewallApplicationRule.kql

```diff
@@ -0,0 +1,38 @@
+let applicationRuleAllowsHttp = AzureDiagnostics
+| where ResourceType == "AZUREFIREWALLS"
+| where Category == "AzureFirewallApplicationRule"
+| parse msg_s with * " request from " ip_source ":" port_source " to " ip_destination ":" port_destination ". Url: " url ". Action: " action ". Policy: " policy ". Rule Collection Group: " rulecollectiongroup "Rule Collection: " rulecollection ". Rule: " rule
+| where action == "Allow"
+| project-away msg_s;
+let applicationRuleAllowsHttps = AzureDiagnostics
+| where ResourceType == "AZUREFIREWALLS"
+| where Category == "AzureFirewallApplicationRule"
+| parse msg_s with * " request from " ip_source ":" port_source " to " ip_destination ":" port_destination ". Action: " action ". Policy: " policy ". Rule Collection Group: " rulecollectiongroup "Rule Collection: " rulecollection ". Rule: " rule
+| where action == "Allow" and port_destination !hasprefix "80" and rule !has "Web Category"
+| project-away msg_s;
+let applicationRuleAllowsHttpsWebCategory = AzureDiagnostics
+| where ResourceType == "AZUREFIREWALLS"
+| where Category == "AzureFirewallApplicationRule"
+| parse msg_s with * " request from " ip_source ":" port_source " to " ip_destination ":" port_destination ". Action: " action ". Policy: " policy ". Rule Collection Group: " rulecollectiongroup "Rule Collection: " rulecollection ". Rule: " rule ". Web Category: " webcategory
+| where action == "Allow"
+| project-away msg_s;
+let applicationRuleDeniesHttp = AzureDiagnostics
+| where ResourceType == "AZUREFIREWALLS"
+| where Category == "AzureFirewallApplicationRule"
+| parse msg_s with * " request from " ip_source ":" port_source " to " ip_destination ":" port_destination ". Url: " url ". Action: " action ". " rule ". " *
+| where action == "Deny" and port_destination == 80

```

Comment:
Would only rules for port 80 log URL? Doesn't the firewall know how to handle other ports as HTTP (or even HTTPS)?

Reply:
You're right. I've taken a better look at how all the entries differ and create a better version with parse-where for each of the event types without looking to much at specific protocol of port details.